aiomusiccast/capability_registry.py

```diff
@@ -0,0 +1,260 @@
+        lambda val: device.set_dts_dialogue_control(zone_id, int(val)),
+        device.data.zones[zone_id].range_step["dts_dialogue_control"].minimum,
+        device.data.zones[zone_id].range_step["dts_dialogue_control"].maximum,
+        device.data.zones[zone_id].range_step["dts_dialogue_control"].step,
+    ),
+    ZoneFeature.LINK_AUDIO_DELAY: lambda capability_id, device, zone_id: OptionSetter(
+        capability_id,
+        "Link Audio Delay",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].link_audio_delay,
+        lambda val: device.set_link_audio_delay(zone_id, val),
+        {key: key for key in device.data.zones[zone_id].link_audio_delay_list},
+    ),
+    ZoneFeature.LINK_CONTROL: lambda capability_id, device, zone_id: OptionSetter(
+        capability_id,
+        "Link Control",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].link_control,
+        lambda val: device.set_link_control(zone_id, val),
+        {key: key for key in device.data.zones[zone_id].link_control_list},
+    ),
+    ZoneFeature.LINK_AUDIO_QUALITY: lambda capability_id, device, zone_id: OptionSetter(
+        capability_id,
+        "Link Audio Quality",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].link_audio_quality,
+        lambda val: device.set_link_audio_quality(zone_id, val),
+        {key: key for key in device.data.zones[zone_id].link_audio_quality_list},
+    ),
+    ZoneFeature.BASS_EXTENSION: lambda capability_id, device, zone_id: BinarySetter(
+        capability_id,
+        "Bass Extension",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].bass_extension,
+        lambda val: device.set_bass_extension(zone_id, val),
+    ),
+    ZoneFeature.EXTRA_BASS: lambda capability_id, device, zone_id: BinarySetter(
+        capability_id,
+        "Extra Bass",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].extra_bass,
+        lambda val: device.set_extra_bass(zone_id, val),
+    ),
+    ZoneFeature.ENHANCER: lambda capability_id, device, zone_id: BinarySetter(
+        capability_id,
+        "Enhancer",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].enhancer,
+        lambda val: device.set_enhancer(zone_id, val),
+    ),
+    ZoneFeature.PURE_DIRECT: lambda capability_id, device, zone_id: BinarySetter(
+        capability_id,
+        "Pure Direct",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].pure_direct,
+        lambda val: device.set_pure_direct(zone_id, val),
+    ),
+    ZoneFeature.ADAPTIVE_DRC: lambda capability_id, device, zone_id: BinarySetter(
+        capability_id,
+        "Adaptive DRC",
+        EntityTypes.CONFIG,
+        lambda: device.data.zones[zone_id].adaptive_drc,
+        lambda val: device.set_adaptive_drc(zone_id, val),
+    ),
+}
+
+
+def build_device_capabilities(device: "MusicCastDevice") -> list[Capability]:
```

Comment:
```suggestion
def build_device_capabilities(device: MusicCastDevice) -> list[Capability]:
```

Reply:
This causes circular imports. For that reason I added the type hint as string only